Fix support for readOnly on all RA Inputs (#9656)
marmelab/react-admin: docs/ReferenceArrayInput.md

<ReferenceArrayInput source="tags_ids" reference="tags" />
```

## `readOnly`

The `readOnly` prop set to true makes the element not mutable, meaning the user can not edit the control.

```jsx
<ReferenceArrayInput
    source="tags_ids"
    reference="tags"
    readOnly
/>
```

> **Warning**: The `readOnly` prop refers to the default [`<AutocompleteArrayInput>`](./AutocompleteArrayInput.md) child. If you pass a child component, you'll have to specify the `readOnly` prop on each input:

Reviewer: <!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**nitpick (non-blocking):** `ReferenceArrayInput‏` accepts at most 1 child I believe

```suggestion
**Warning**: The `readOnly` prop is forwarded to the default [`<AutocompleteArrayInput>`](./AutocompleteArrayInput.md) child. If you pass a custom child component, you'll have to specify the `readOnly` prop on it instead:
```

```jsx
<ReferenceArrayInput source="tags_ids" reference="tags">
    <SelectArrayInput readOnly />
</ReferenceArrayInput>
```

## `disabled`

The `disabled` prop set to true makes the element not mutable, focusable, or even submitted with the form.

```jsx
<ReferenceArrayInput
    source="tags_ids"
    reference="tags"
    disabled
/>
```

**Tip:** Note that `disabled` inputs are **not** included in the form values, and hence may trigger `warnWhenUnsavedChanges` if the input previously had a value in the record. To include the input in the form values, you can use `readOnly` instead of `disabled`.

> **Warning**: The `disabled` prop refers to the default [`<AutocompleteArrayInput>`](./AutocompleteArrayInput.md) child. If you pass a child component, you'll have to specify the `disabled` prop on each input:

Reviewer: ```suggestion
**Warning**: The `disabled` prop is forwarded to the default [`<AutocompleteArrayInput>`](./AutocompleteArrayInput.md) child. If you pass a custom child component, you'll have to specify the `disabled` prop on it instead:
```

<ReferenceArrayInput source="tags_ids" reference="tags" />
```

## `readOnly`

> If `true`, the default input is in read-only mode and the user can't change the value.

Reviewer 2: <!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏I think something like this could help users to understand the differences between `readOnly` and `disabled` props:
```suggestion
The `readOnly` prop set to true makes the element not mutable, meaning the user can not edit the control.
```

> {% raw %}

Reviewer 2: <!-- Polish comments are like a suggestion, where there is nothing necessarily wrong with the relevant content, there's just some ways to immediately improve the quality.  -->
**polish:** ‏ `{% raw %}` are not necessary when there is no objects inside the snippet. We should remove them when they are not needed.

```jsx
<ReferenceArrayInput
    source="tags_ids"
    reference="tags"
    readOnly
/>
```

{% endraw %}

> **Warning**: The `readOnly` prop refer to the default [`<AutocompleteArrayInput>`](./AutocompleteArrayInput.md) child. If you defined `children` you have to specify the `readOnly` prop on each input:

Reviewer 2: <!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏
```suggestion
**Warning**: The `readOnly` prop refers to the default [`<AutocompleteArrayInput>`](./AutocompleteArrayInput.md) child. If you pass a child component, you'll have to specify the `readOnly` prop on each input:
```

{% raw %}

```jsx
<ReferenceArrayInput source="tags_ids" reference="tags">
    <SelectArrayInput readOnly />
</ReferenceArrayInput>
```

{% endraw %}

## `disabled`

> If `true`, the default input is disabled and the user can't change the value.

Reviewer 2: **suggestion**: Same as before, ‏I think something like this could help users to understand the differences between readOnly and disabled props:
```suggestion
The `disabled` prop set to true makes the element not mutable, focusable, or even submitted with the form. The user can neither edit nor focus on the control, nor its form control descendants.
```

{% raw %}

```jsx
<ReferenceArrayInput
    source="tags_ids"
    reference="tags"
    disabled
/>
```

{% endraw %}

> **Tip**: The form framework used by react-admin, react-hook-form, [considers](https://github.com/react-hook-form/react-hook-form/pull/10805) that a `disabled` input shouldn't submit any value. So react-hook-form sets the value of all `disabled` inputs to `undefined`. As a consequence, a form with a `disabled` input is always considered `dirty` (i.e. react-hook-form considers that the form values and the initial record values are different), and it triggers [the `warnWhenUnsavedChanges` feature](./EditTutorial.md#warning-about-unsaved-changes) when leaving the form, even though the user changed nothing. The workaround is to set the `disabled` prop on the underlying input component, as follows:

Reviewer 2: <!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought (non-blocking):** I'm not sure about adding this paragraph. It seems like we are blaming RHF, but they are following the [readonly specs](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled). What do you think @slax57

{% raw %}

```jsx
<ReferenceArrayInput
    source="tags_ids"
    reference="tags"
    disabled
/>
```

{% endraw %}

> **Warning**: The `disabled` prop refer to the default [`<AutocompleteArrayInput>`](./AutocompleteArrayInput.md) child. If you defined `children` you have to specify the `disabled` prop on each input:

Reviewer 2: ```suggestion
**Warning**: The `disabled` prop refers to the default [`<AutocompleteArrayInput>`](./AutocompleteArrayInput.md) child. If you pass a child component, you'll have to specify the `disabled` prop on each input:
```
